### DISCUSSION TITLE
API documentation needed!

### DISCUSSION THREAD

**Discussion by FlyinCow** [fastapi/fastapi#8742] [Questions] [answered]
People NEED a DETAILED (maybe indexed, too) API documentation, like [python documentation](https://docs.python.org/3/). But it seems we only get a tutorial.

**Comment by ArcLightSlavik** [comment]
Duplicate of #804 and #2275

**Comment by ricardo-reis-1970** [comment]
This is a great topic. An API Reference is lacking, as many times all us devs need is a quick fingerprint of a particular function and we don't have one. I would suggest something like `pydoc` but apparently there are not a lot of docstrings in the code.

And **YES** @tiangolo, we all know this is a huge effort for an opensource project, but I truly believe that the API Reference is absolutely necessary. It won't make FastAPI great (because it already is), but it will make it more adoptable by the not-so-stubborn among us.

**Comment by YuriiMotov** [comment, marked as answer]
https://fastapi.tiangolo.com/reference/
